internal/component/mimir/rules/kubernetes/events_test.go

```diff
@@ -393,6 +392,208 @@ func TestExtraQueryMatchers(t *testing.T) {
+	nsIndexer := testNamespaceIndexer()
+	ruleIndexer := testRuleIndexer()
+
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "namespace",
+			UID:  types.UID("33f8860c-bd06-4c0d-a0b1-a114d6b9937b"),
+		},
+	}
+
+	rule := &v1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "name",
+			Namespace: "namespace",
+			UID:       types.UID("64aab764-c95e-4ee9-a932-cd63ba57e6cf"),
+			Annotations: map[string]string{
+				AnnotationsSourceTenants: "tenant1,tenant2",
+			},
+		},
+		Spec: v1.PrometheusRuleSpec{
+			Groups: []v1.RuleGroup{
+				{
+					Name: "group1",
+					Rules: []v1.Rule{
+						{
+							Alert: "alert1",
+							Expr:  intstr.FromString("expr1"),
+						},
+						{
+							Alert: "alert2",
+							Expr:  intstr.FromString("expr2"),
+							Labels: map[string]string{
+								// This label should get overridden.
+								"foo": "lalalala",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	processor := &eventProcessor{
+		queue:             workqueue.NewTypedRateLimitingQueue[kubernetes.Event](workqueue.DefaultTypedControllerRateLimiter[kubernetes.Event]()),
+		stopChan:          make(chan struct{}),
+		health:            &fakeHealthReporter{},
+		mimirClient:       newFakeMimirClient(),
+		namespaceLister:   coreListers.NewNamespaceLister(nsIndexer),
+		ruleLister:        promListers.NewPrometheusRuleLister(ruleIndexer),
+		namespaceSelector: labels.Everything(),
+		ruleSelector:      labels.Everything(),
+		namespacePrefix:   "alloy",
+		metrics:           newMetrics(),
+		logger:            log.With(log.NewLogfmtLogger(os.Stdout), "ts", log.DefaultTimestampUTC),
+	}
+
+	ctx := t.Context()
+
+	// Do an initial sync of the Mimir ruler state before starting the event processing loop.
+	require.NoError(t, processor.syncMimir(ctx))
+	go processor.run(ctx)
+	defer processor.stop()
+
+	eventHandler := kubernetes.NewQueuedEventHandler(processor.logger, processor.queue)
+
+	// Add a namespace and rule to kubernetes
+	require.NoError(t, nsIndexer.Add(ns))
+	require.NoError(t, ruleIndexer.Add(rule))
+	eventHandler.OnAdd(rule, false)
+
+	// Wait for the rule to be added to mimir
+	rules := map[string][]client.MimirRuleGroup{}
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		var err error
+		rules, err = processor.mimirClient.ListRules(ctx, "")
+		assert.NoError(c, err)
+		assert.Len(c, rules, 1)
+	}, 10*time.Second, 10*time.Millisecond)
+
+	// The map of rules has only one element.
+	for ruleName, rule := range rules {
+		require.Equal(t, "alloy/namespace/name/64aab764-c95e-4ee9-a932-cd63ba57e6cf", ruleName)
+
+		ruleBuf, err := yaml.Marshal(rule)
+		require.NoError(t, err)
+
+		expectedRule := `- name: group1
+  rules:
+  - alert: alert1
+    expr: expr1
+  - alert: alert2
+    expr: expr2
+    labels:
+      foo: lalalala
+  source_tenants: ["tenant1","tenant2"]
+`
+		require.YAMLEq(t, expectedRule, string(ruleBuf))
+	}
+}
+
+func TestSourceTenants(t *testing.T) {
```

Comment:
There are two test with the same name and I think they are identical so you need to remove one of them.